### ISSUE TITLE
Readme says server sleeps

### ISSUE THREAD

**Issue by mcstoer** [dbakewel/netbots#52]
Should this line be changed as the server no longer truly sleeps?
`If the Step/Message Loop takes less time then the server will sleep until the next loop is scheduled to start.`

**Comment by dbakewel** [comment]
Well the server does still sleep. It just does it with it's own busy wait rather than asking the OS to do it. During server sleep the same thing still happens: the server does not do anything. Does that make sense?

**Comment by mcstoer** [comment]
For the sake of wording, maybe change sleep to wait?

**Comment by dbakewel** [comment]
Actually I think the sleep is the correct term. Sleep implies waiting on a time event. Wait implies it's waiting on some other type of interrupt or message to be received or something like that. That's just the way developers tend to use those terms. Sleep may not make much sense but keep in mind a computer is always busy waiting at some level whether it's in the software, the operating system, or the hardware. Computers are not truly capable of stopping. We still call that sleeping if it's not doing anything and waiting on a timed event.

**Comment by dbakewel** [comment]
This is a short outline of how 'sleep' and 'wait' terms are generally used in CS.

> **Wait** : This system call is used when user wants to halt the current process and it **releases all the resources** held by the process and waits for some other process to execute. We need to **use notify** to make this process aware that start execution again.

> **Sleep**: This system call is used when user wants to halt the current process **for sometime**. It keep the locks hold on resources till the sleep time is over and again starts the execution of the process. Here **process has control throughout the execution**. 

Cut from: https://www.quora.com/What-is-the-difference-between-wait-and-sleep-in-operating-system

**Comment by mcstoer** [comment]
From Wikipedia Sleep (system call)
https://en.wikipedia.org/wiki/Sleep_(system_call),

> A computer program (process, task, or thread) may sleep, which places it into an **inactive state** for a period of time.

The server process is not inactive during the period where it is not updating a step.
The server acts more like a wait where the

>... function blocks the calling process until one of its child processes exits or a signal is received.

from http://www.csl.mtu.edu/cs4411.ck/www/NOTES/process/fork/wait.html

As the while loop that constantly checks the time acts as a child process, where the updating of a step is the parent/main process.

I think the wording should be changed to
` ... the server will busy wait ... `
for the most clarity.
